### PR TITLE
Small builds changes for end to end source build

### PR DESCRIPTION
**Pull request by ellismg**
Here's a small set of changes I've added recently for the end to end source build effort.  Mainly around adding some additional flexibility to support skipping some things we don't care about during the Build phase.

/cc @jaredpar

### CODE REVIEW

File: cibuild.cmd [hunk 1]
```diff
@@ -22,6 +23,8 @@ if /I "%1" == "/testBuildCorrectness" set TestBuildCorrectness=true&&shift&& got
 if /I "%1" == "/testPerfCorrectness" set TestPerfCorrectness=true&&shift&& goto :ParseArguments
 if /I "%1" == "/testPerfRun" set TestPerfRun=true&&shift&& goto :ParseArguments
 if /I "%1" == "/testVsi" set TestVsi=true&&shift&& goto :ParseArguments
+if /I "%1" == "/skipTests" set BuildAndTestBuildTarget=Build&&shift&&goto :ParseArguments
```
Comment (jaredpar) [inline cibuild.cmd]: Please make it skipTest (singular).  The singular form is used consistently in our repo.  Example.

https://github.com/dotnet/roslyn/blob/master/src/Tools/MicroBuild/Build.proj#L40

File: cibuild.cmd [hunk 2]
```diff
@@ -9,6 +9,7 @@ REM Because override the C#/VB toolset to build against our LKG package, it is i
 REM that we do not reuse MSBuild nodes from other jobs/builds on the machine. Otherwise,
 REM we'll run into issues such as https://github.com/dotnet/roslyn/issues/6211.
 set MSBuildAdditionalCommandLineArgs=/nologo /m /nodeReuse:false /consoleloggerparameters:Verbosity=minimal /filelogger /fileloggerparameters:Verbosity=normal
+set BuildAndTestBuildTarget=BuildAndTest
```
Comment (jaredpar) [inline cibuild.cmd]: Shouldn't the default be Build?

Comment (jaredpar) [inline cibuild.cmd]: This is never changed ... unsure why you made this change.

Comment (ellismg) [inline cibuild.cmd]: It's changed to just "Build" if "/SkipTest" is passed.

File: cibuild.cmd [hunk 3]
```diff
@@ -115,7 +120,7 @@ if defined TestPerfRun (
     exit /b 0
 )
 
-msbuild %MSBuildAdditionalCommandLineArgs% /p:BootstrapBuildPath="%bindir%\Bootstrap" BuildAndTest.proj /p:Configuration=%BuildConfiguration% /p:Test64=%Test64% /p:TestVsi=%TestVsi% /p:RunProcessWatchdog=%RunProcessWatchdog% /p:BuildStartTime=%BuildStartTime% /p:"ProcDumpExe=%ProcDumpExe%" /p:BuildTimeLimit=%BuildTimeLimit% /p:PathMap="%RoslynRoot%=q:\roslyn" /p:Feature=pdb-path-determinism /fileloggerparameters:LogFile="%bindir%\Build.log";verbosity=diagnostic /p:DeployExtension=false || goto :BuildFailed
+msbuild %MSBuildAdditionalCommandLineArgs% /p:BootstrapBuildPath="%bindir%\Bootstrap" BuildAndTest.proj /t:%BuildAndTestBuildTarget% /p:Configuration=%BuildConfiguration% /p:Test64=%Test64% /p:TestVsi=%TestVsi% /p:RunProcessWatchdog=%RunProcessWatchdog% /p:BuildStartTime=%BuildStartTime% /p:"ProcDumpExe=%ProcDumpExe%" /p:BuildTimeLimit=%BuildTimeLimit% /p:PathMap="%RoslynRoot%=q:\roslyn" /p:Feature=pdb-path-determinism /fileloggerparameters:LogFile="%bindir%\Build.log";verbosity=diagnostic /p:DeployExtension=false || goto :BuildFailed
```
Comment (ellismg) [inline cibuild.cmd]: @jaredpar Note the addition of `/t:%BuildAndTestBuildTarget%` here which uses the new value we set above (and will change if `/skiptest` is set)

File: cibuild.cmd (hunk 1 above)
Comment (ellismg) [inline cibuild.cmd]: Will do!

Comment (ellismg) [inline cibuild.cmd]: Added 0864abf and fixed the switch to have the same name in `cibuild.sh` (I added that switch back a while back).

**Comment by ellismg** [comment]
@jaredpar It looks like the perf_correctness_prtest leg [failed](https://ci.dot.net/job/dotnet_roslyn/job/master/job/perf_correctness_prtest/2029/console) because it couldn't kill MSBuild.  I touched the invocation of taskkill so this could be my fault, but I was wondering if you had seen this before?

**Comment by jaredpar** [comment]
CC @TyOverby who would know why that happens.

**Comment by TyOverby** [comment]
I can't think of any reason that this would fail here :(  

It could be that the last job on the machine failed to clean up. (it would fail that build as well)

**Comment by ellismg** [comment]
@dotnet-bot test perf_correctness_prtest please

**Comment by TyOverby** [comment]
@jaredpar @ellismg: looks like that was the case :(

**Pull request merged**